### DISCUSSION TITLE
Ollama model support

### DISCUSSION THREAD

**Discussion by swuecho** [swuecho/chat#396] [Announcements]
1. download ollama and pull mistral model.
```
curl -fsSL https://ollama.com/install.sh | sh
ollama pull mistral
```

![image](https://github.com/swuecho/chat/assets/666683/3695c088-4dcd-4ff4-9a75-6b9d44186a4b)

3. config ollama model in chat admin

<img width="1293" alt="image" src="https://github.com/swuecho/chat/assets/666683/65cfca8a-edbf-4fa2-844c-c3b536b32cd0">

id: ollama-{modelName}
name: does not matter, naming as you like
baseUrl: http://hostname:11434/api/chat
other fields is irrelevant.

enjoy!

**Comment by swuecho** [comment]
<img width="1116" alt="image" src="https://github.com/swuecho/chat/assets/666683/8064b861-3089-47a7-8331-fc714ab4fe98">

**Comment by swuecho** [comment]
![image](https://github.com/swuecho/chat/assets/666683/e301f393-157c-46ea-9eb9-3c4f0de0c7ca)

**Comment by cyx2000** [comment]
ollama的接口是不是写错了，用的是chat接口的参数，接口换成chat的就可以使用，generate不行
![image](https://github.com/swuecho/chat/assets/56961601/abfe3852-6c42-4b97-9bdc-ba69bfd929f1)
generate接口的参数
![image](https://github.com/swuecho/chat/assets/56961601/a690de23-02b7-4067-85a2-bc49e6a84f1e)

**Reply by swuecho** [reply]
是chat

最初ollama 没有chat api

**Reply by cyx2000** [reply]
哦哦，那文档要更新下
